### PR TITLE
Move benchmarks to separate module

### PR DESCRIPTION
**Pull request by Munzey**
Follow on from  #41

Made the benchmark class fully multiplatform compliant and added the js target.
Unfortuantely the benchmarking plugin hasda couple of weird issues to get the js target working

### CODE REVIEW

File: benchmarks/src/commonMain/kotlin/com/github/michaelbull/result/BindingBenchmark.kt [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+import kotlinx.benchmark.BenchmarkMode
+import kotlinx.benchmark.Mode
+import kotlinx.benchmark.OutputTimeUnit
+import kotlinx.benchmark.BenchmarkTimeUnit
+import kotlinx.benchmark.Benchmark
+import kotlinx.benchmark.Blackhole
+import kotlinx.benchmark.State
+import kotlinx.benchmark.Scope
 
+@State(Scope.Benchmark)
```
Comment (Munzey) [inline benchmarks/src/commonMain/kotlin/com/github/michaelbull/result/BindingBenchmark.kt:12]: For some reason, for js benchmark target to pick up any tests it needs the whole class annotated, opened a ticket: https://github.com/Kotlin/kotlinx-benchmark/issues/29

File: buildSrc/src/main/kotlin/Versions.kt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 object Versions {
     const val dokka = "0.10.1"
-    const val kotlin = "1.4.21-2"
+    const val kotlin = "1.4.10"
```
Comment (Munzey) [inline buildSrc/src/main/kotlin/Versions.kt:3]: js benchmarking failing to compile with anything higher, opened a ticket: https://github.com/Kotlin/kotlinx-benchmark/issues/28

**Comment by michaelbull** [comment]
Can we try disabling the JS benchmarks instead of downgrading Kotlin to satisfy them being broken?

File: buildSrc/src/main/kotlin/Versions.kt (hunk 2 above)
Comment (Munzey) [reply buildSrc/src/main/kotlin/Versions.kt]: disabled js benchmark target and bumped to 1.4.30

**Comment by michaelbull** [comment]
Merged in 0df4c62d4f755d8e60197ad46d28bbf658c8173f

**Pull request closed without merging**